x-pack/platform/plugins/shared/actions/server/lib/token_tracking/get_token_count_from_bedrock_converse_stream.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type { Readable } from 'stream';
+import { EventStreamMarshaller } from '@smithy/eventstream-serde-node';
+import { fromUtf8, toUtf8 } from '@smithy/util-utf8';
+import { identity } from 'lodash';
+import { finished } from 'stream/promises';
+
+export const getTokensFromBedrockConverseStream = async function (
+  responseStream: Readable,
+  logger: Logger
+): Promise<{ total_tokens: number; prompt_tokens: number; completion_tokens: number } | null> {
+  try {
+    const marshaller = new EventStreamMarshaller({
+      utf8Encoder: toUtf8,
+      utf8Decoder: fromUtf8,
+    });
+    const responseBuffer: unknown[] = [];
+    for await (const chunk of marshaller.deserialize(responseStream, identity)) {
+      if (chunk) {
+        responseBuffer.push(chunk);
+      }
+    }
+    try {
+      await finished(responseStream);
+    } catch (e) {
+      logger.error('An error occurred while calculating streaming response tokens');
```

Comment:
Can we log out the full error here in case we need to troubleshoot when an error occurs

Reply:
https://github.com/elastic/kibana/pull/227531/commits/c96dea4dd8307bb2c5942c9c4f0d1cd707eaff22